src/content/hinting.ts

```diff
@@ -1093,61 +1102,44 @@ export function hintableImages() {
 export function hintByText(match: string | RegExp) {
     return DOM.getElemsBySelector(DOM.HINTTAGS_filter_by_text_selectors, [
         DOM.isVisible,
-        hint => {
-            let text
-            if (hint instanceof HTMLInputElement) {
-                // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
-                text = (hint as HTMLInputElement).value
-            } else {
-                text = hint.textContent
-            }
-            if (match instanceof RegExp) {
-                return text.match(match) !== null
-            } else {
-                return text.toUpperCase().includes(match.toUpperCase())
-            }
-        },
+        hintByTextFilter(match),
     ])
 }
 
+/** Return a predicate that checks whether an element matches a given text hinting filter
+ * @hidden
+ */
+export function hintByTextFilter(match: string | RegExp): HintSelectedCallback {
+    return hint => {
+        let text
+        if (hint instanceof HTMLInputElement) {
+            // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+            text = (hint as HTMLInputElement).value
```

Comment:
```suggestion
            text = hint.value
```

Reply:
Good catch, I didn't really look at this and just moved the filter's code into its own function!